nautobot/extras/api/serializers.py

```diff
@@ -766,6 +767,36 @@ class Meta:
         fields = "__all__"
 
 
+class ObjectMetadataSerializer(ValidatedModelSerializer):
+    assigned_object_type = ContentTypeField(
+        queryset=ContentType.objects.filter(FeatureQuery("metadata").get_query()),
+    )
+    assigned_object = serializers.SerializerMethodField()
+
+    class Meta:
+        model = ObjectMetadata
+        fields = "__all__"
```

Comment:
Do we need a special field for `value` now?

Reply:
Is there something similar with CustomField? I am trying to find it with no avail.

Reply:
Not that I'm aware of. For CustomField values on an object, it's just a JSON field - might be able to implement something similar for API representation of `value`, though the Contact/Team cases might be easier to represent as distinct `contact` and `team` fields in the API for now rather than trying to have a single `value` that could be a JSON object or a FK reference?

Reply:
I see, I was overthinking it. Declaring `value = serializers.JSONField(read_only=True, required=False)` seems to work just fine in the API.

Reply:
Does that work when value is a Contact or Team?

Reply:
Also, shouldn't need to be `read_only=True` once the property setter is implemented, maybe? Definitely need the ability to set `value` via REST.